src/native/credentials_provider.c

```diff
@@ -379,7 +379,11 @@ static int s_credentials_provider_delegate_get_credentials(
 
     struct aws_byte_cursor access_key_id = aws_jni_byte_cursor_from_jbyteArray_acquire(env, java_access_key_id);
     struct aws_byte_cursor secret_access_key = aws_jni_byte_cursor_from_jbyteArray_acquire(env, java_secret_access_key);
-    struct aws_byte_cursor session_token = aws_jni_byte_cursor_from_jbyteArray_acquire(env, java_session_token);
+    struct aws_byte_cursor session_token;
```

Comment:
Is it worth doing similar checks for the access_key_id and the secret_access_key?  (Maybe not the same exact behavior, but something that prevents a crash if there's a null.)

Reply:
Good call! Should throw an exception if one of them is null. Updated